src/types/permissions-types.ts

```diff
@@ -9,9 +9,11 @@ export type PermissionScope = {
 // Method-specific scopes
 export type RecordsPermissionScope = {
   interface: DwnInterfaceName.Records;
-  method: DwnMethodName.Read | DwnMethodName.Write | DwnMethodName.Delete;
-  protocol?: string;
-  schema?: string;
+  method: DwnMethodName.Read | DwnMethodName.Write;
+  protocol?: string; // May only be present when `schema` is undefined
+  contextId?: string; // May only be present when `protocol` is defined and `protocolPath` is undefined
+  protocolPath?: string; // May only be present when `protocol` is defined and `contextId` is undefined
+  schema?: string; // May only be present when `protocol` is undefined
```

Comment:
Might as well use `/** ... */ notation for the property comments so that the comments are picked up by IntelliSense and doc generator.

Reply:
Updated